### DISCUSSION TITLE
Is `sidekiq_options` safe to use for custom settings for custom middleware?

### DISCUSSION THREAD

**Discussion by tony-pizza** [sidekiq/sidekiq#5950] [Q&A] [answered]
I'm tempted to use `sidekiq_options` to set configuration for custom middleware.

To illustrate, consider configuring a worker to use some custom monitoring middleware that requires a `monitor_id` setting per worker:

```ruby
class MyWorker
  include Sidekiq::Worker
  sidekiq_options queue: :high_priority, monitor_id: "abc123"
  # ...
end

# vs

class MyWorker
  include Sidekiq::Worker
  include MonitoredWorker
  sidekiq_options queue: :high_priority
  set_monitor_id "abc123"
  # ...
end
```

 I like this pattern because...
- it's consistent with Sidekiq's own middlewares (e.g. [Expiring Jobs adds an `expires_in` setting](https://github.com/sidekiq/sidekiq/wiki/Pro-Expiring-Jobs))
- it could allow per-job override behavior when pushing to Sidekiq via `MyWorker.set(my_setting: "abc").perform_async` (**EDIT:** I haven't tried this, but I assume this would require that the setting be safely serialized into Redis)
- it makes it obvious the setting is Sidekiq-related
- it's fewer lines of worker code compared to other approaches

Looking at the source, it seems pretty safe and straightforward to have middleware use custom fields in `sidekiq_options`. However, I haven't found any documentation on using `sidekiq_options` for custom middleware settings. Are there reasons you'd advise against using `sidekiq_options` in this way?

**Comment by mperham** [comment, marked as answer]
Yes, that's absolutely idiomatic and recommended usage.
